integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,423 @@
+		
+		// verify created correctly
+		assertThat(response, isCreated());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Task task = readResponse(response);
+		
+		assertThat(task, notNullValue());
+		assertThat(task.getIdElement().getIdPart(), notNullValue());
+		assertThat(task.getAuthoredOn(), equalTo("2016-10-31T08:25:05+10:00"));
+		assertThat(task.getStatus(), equalTo("ACCEPTED"));
+		assertThat(task.getIntent(), equalTo("order"));
+		assertThat(task, validResource());
+		
+		assertThat(task.getOwner(), equalTo("owner Requested Task"));
+		assertThat(task, validResource());
+		
+		// try to get new task
+		response = get("/Task/" + task.getIdElement().getIdPart()).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		
+		Task newTask = readResponse(response);
+		
+		assertThat(newTask.getId(), equalTo(task.getId()));
+		assertThat(newTask.getIdElement().getIdPart(), notNullValue());
+	}
+	
+	@Test
+	public void shouldCreateNewTaskAsXML() throws Exception {
+		String xmlTask;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(XML_CREATE_TASK_DOCUMENT)) {
+			Objects.requireNonNull(is);
+			xmlTask = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		// create test
+		MockHttpServletResponse response = post("/Task").accept(FhirMediaTypes.XML).xmlContext(xmlTask).go();
+		
+		// verify created correctly
+		assertThat(response, isCreated());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Task task = readResponse(response);
+		
+		assertThat(task, notNullValue());
+		assertThat(task.getIdElement().getIdPart(), notNullValue());
+		assertThat(task.getAuthoredOn(), equalTo("2016-10-31T08:25:05+10:00"));
+		assertThat(task.getStatus(), equalTo("ACCEPTED"));
+		assertThat(task.getIntent(), equalTo("order"));
+		assertThat(task, validResource());
+		
+		assertThat(task.getOwner(), equalTo("owner Requested Task"));
+		assertThat(task, validResource());
+		
+		// try to get new task
+		response = get("/Task/" + task.getIdElement().getIdPart()).accept(FhirMediaTypes.XML).go();
+		
+		assertThat(response, isOk());
+		
+		Task newTask = readResponse(response);
+		
+		assertThat(newTask.getId(), equalTo(task.getId()));
+		assertThat(newTask.getIdElement().getIdPart(), notNullValue());
+		
+	}
+	
+	@Test
+	public void shouldUpdateExistingTaskAsJson() throws Exception {
+		MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		Task task = readResponse(response);
+		
+		// update the existing record
+		Date authoredOn = DateUtils.truncate(new Date(), Calendar.DATE);
+		task.setAuthoredOn(authoredOn);
+		
+		// send the update to the server
+		response = put("/Task/" + TASK_UUID).jsonContent(toJson(task)).jsonContent(toJson(task)).go();
```

Comment:
```suggestion
		response = put("/Task/" + TASK_UUID).jsonContent(toJson(task)).go();
```

Reply:
This brings in errors

Reply:
Yes, but that's the correct way to go about it. Any particular reason for putting the same content twice? It doesn;t make sense to me.

Reply:
Resolved thanks